### PR TITLE
[IOTDB-2945] Reconstruct ConfigNode manage layer and persistence layer

### PR DESCRIPTION
**Pull request by CRZbulabula**
Currently, ConfigNode manages Region allocation information and StorageGroupSchema in RegionInfoPersistence. But ConfigNode must prepare to storage cluster schema, and the StorageGroupSchema belongs to the cluster schema. So this refactoring normalized the ConfigNode's manage layer and persistence layer.

Changes:

- Rename and move ConfigNode inner classes
- Remove RegionInfoPersistence
- Add ClusterSchemaManager
- Add StorageGroupInfo
- Merge RegionInfoPersistence and PartitionInfoPersistence
- Split RegionManager into PartitionManager and ClusterSchemaManager

See [ConfigNode 架构](https://y8dp9fjm8f.feishu.cn/docs/doccnUzGLXivqOoNjAo8Ole3cQe) for more details.

### CODE REVIEW

**Comment by CRZbulabula** [comment]
WIP.

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java [hunk 1]
```diff
@@ -56,40 +68,81 @@ public class PartitionInfoPersistence {
...
+  /**
+   * Persistence allocation result of new Regions
+   *
+   * @param plan CreateRegionsPlan
+   * @return SUCCESS_STATUS
+   */
+  public TSStatus createRegions(CreateRegionsReq plan) {
+    TSStatus result;
+    regionReadWriteLock.writeLock().lock();
+    regionAllocateLock.writeLock().lock();
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java:112]: why does add a regionAllocateLock?     
can regionReadWriteLock resoved only?

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java [hunk 2]
```diff
@@ -56,40 +68,81 @@ public class PartitionInfoPersistence {
...
             ConfigNodeDescriptor.getInstance().getConf().getSeriesPartitionExecutorClass(),
             ConfigNodeDescriptor.getInstance().getConf().getSeriesPartitionSlotNum());
     this.dataPartition.setDataPartitionMap(new HashMap<>());
   }
 
+  public int generateNextRegionGroupId() {
+    int result;
+    regionAllocateLock.writeLock().lock();
+    try {
+      result = nextRegionGroupId;
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java:95]: if nextRegionGroupId  use AtomicLong,  will be not need lock...

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java (hunk 1 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java]: Fixed~

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java (hunk 2 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java]: Fixed~

**Review by RYH61 — APPROVED** [review]

**Review by wangchao316 — APPROVED** [review]

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ClusterSchemaManager.java [hunk 3]
```diff
@@ -117,11 +113,9 @@ public TSStatus setStorageGroup(SetStorageGroupPlan plan) {
...
   }
 
-  private DataNodeManager getDataNodeInfoManager() {
-    return configNodeManager.getDataNodeManager();
-  }
-
-  private void allocateRegions(TConsensusGroupType type, CreateRegionsPlan plan) {
+  /** TODO: Allocate by LoadManager */
+  private void allocateRegions(
+      TConsensusGroupType type, CreateRegionsReq createPlan, SetStorageGroupReq setPlan) {
```
Comment (qiaojialin) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/ClusterSchemaManager.java:118]:
```suggestion
      TConsensusGroupType type, CreateRegionsReq createRegionsReq, SetStorageGroupReq setSGReq) {
```

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/Manager.java [hunk 4]
```diff
@@ -90,10 +90,10 @@ public interface Manager {
   /**
    * Set StorageGroup
    *
-   * @param physicalPlan SetStorageGroupPlan
+   * @param configRequest SetStorageGroupPlan
```
Comment (qiaojialin) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/Manager.java:93]:
```suggestion
   * @param configRequest SetStorageGroupReq
```

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java [hunk 5]
```diff
@@ -213,14 +285,14 @@ public void clear() {
 
   private static class PartitionInfoPersistenceHolder {
```
Comment (qiaojialin) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java:286]:
```suggestion
  private static class PartitionInfoHolder {
```

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/StorageGroupInfo.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+import org.apache.iotdb.confignode.rpc.thrift.TStorageGroupSchema;
+import org.apache.iotdb.rpc.TSStatusCode;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class StorageGroupInfo {
```
Comment (qiaojialin) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/StorageGroupInfo.java:36]: We could merge this into ClusterSchemaManager

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ClusterSchemaManager.java (hunk 3 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/ClusterSchemaManager.java]: Fixed~

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/Manager.java (hunk 4 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/Manager.java]: Fixed~

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java (hunk 5 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java]: Fixed~

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/StorageGroupInfo.java (hunk 6 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/StorageGroupInfo.java:36]: The StorageGroup is responsible for persisting StorageGroupSchema. Therefore it cannot be merged with ClusterSchemaManager.

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**